<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MqttMainWindow</class>
 <widget class="QMainWindow" name="MqttMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>979</width>
    <height>485</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MqttMainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>511</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <layout class="QGridLayout" name="gridLayout" columnstretch="0,0,0,0">
     <item row="6" column="1" colspan="3">
      <widget class="QTextEdit" name="teWillMsg"/>
     </item>
     <item row="3" column="2">
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Password:</string>
       </property>
       <property name="buddy">
        <cstring>lePasswd</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QSpinBox" name="sbPort">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>65535</number>
       </property>
       <property name="value">
        <number>1883</number>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="sbKeepalive">
       <property name="suffix">
        <string notr="true">(s)</string>
       </property>
       <property name="maximum">
        <number>3600</number>
       </property>
       <property name="value">
        <number>300</number>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>ClientId:</string>
       </property>
       <property name="buddy">
        <cstring>leClientId</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Username:</string>
       </property>
       <property name="buddy">
        <cstring>leUser</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="leUser"/>
     </item>
     <item row="3" column="3">
      <widget class="QLineEdit" name="lePasswd"/>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="leHost">
       <property name="text">
        <string>localhost</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Will Msg:</string>
       </property>
       <property name="buddy">
        <cstring>teWillMsg</cstring>
       </property>
      </widget>
     </item>
     <item row="4" column="0" colspan="4">
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Port(*):</string>
       </property>
       <property name="buddy">
        <cstring>sbPort</cstring>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Will Topic:</string>
       </property>
       <property name="buddy">
        <cstring>leWillTopic</cstring>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QPushButton" name="connButton">
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Server(*):</string>
       </property>
       <property name="buddy">
        <cstring>leHost</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Keepalive:</string>
       </property>
       <property name="buddy">
        <cstring>sbKeepalive</cstring>
       </property>
      </widget>
     </item>
     <item row="5" column="1" colspan="3">
      <widget class="QLineEdit" name="leWillTopic"/>
     </item>
     <item row="2" column="1" colspan="2">
      <widget class="QLineEdit" name="leClientId">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QCheckBox" name="cbCleanSess">
       <property name="text">
        <string/>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>CleanSess:</string>
       </property>
       <property name="buddy">
        <cstring>cbCleanSess</cstring>
       </property>
      </widget>
     </item>
     <item row="7" column="2">
      <widget class="QPushButton" name="buttonPing">
       <property name="text">
        <string>Ping</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>0</y>
      <width>448</width>
      <height>131</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="6" column="0">
      <widget class="QLabel" name="messageLabel">
       <property name="text">
        <string>Message(*):</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QComboBox" name="qOSComboBox">
       <item>
        <property name="text">
         <string>QOS0</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>QOS1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>QOS2</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="topicLabel">
       <property name="text">
        <string>Topic(*):</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="qOSLabel">
       <property name="text">
        <string>QOS:</string>
       </property>
      </widget>
     </item>
     <item row="6" column="2" colspan="5">
      <widget class="QTextEdit" name="payloadEdit"/>
     </item>
     <item row="2" column="2">
      <widget class="QCheckBox" name="retainCheckBox"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="retainLabel">
       <property name="text">
        <string>Retain:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2" colspan="5">
      <widget class="QLineEdit" name="topicLineEdit"/>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>180</y>
      <width>446</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QLabel" name="topicLabel_2">
         <property name="text">
          <string>Topic:</string>
         </property>
         <property name="buddy">
          <cstring>topicLineEdit</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QComboBox" name="qosComboBox">
         <item>
          <property name="text">
           <string>QOS0</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>QOS1</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>QOS2</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="qosLabel">
         <property name="text">
          <string>Qos:</string>
         </property>
         <property name="buddy">
          <cstring>qosComboBox</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="topicLineEdit_2"/>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Subscribed Topics:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QListWidget" name="listWidget"/>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pubButton">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>130</y>
      <width>75</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>&amp;Publish</string>
    </property>
   </widget>
   <widget class="QPushButton" name="subButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>320</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>&amp;Subscribe</string>
    </property>
   </widget>
   <widget class="QPushButton" name="unsubButton">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>320</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>U&amp;nsubscribe</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="editLog">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>230</y>
      <width>501</width>
      <height>400</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>0</width>
      <height>400</height>
     </size>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>370</y>
      <width>131</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>instances number：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>370</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>370</y>
      <width>171</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string># of messages to subscribe：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_2">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>370</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="loadtestButton">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>410</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>load test</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pubMuiButton">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>130</y>
      <width>75</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>&amp;PubMui</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>979</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
